src/OV/OpenVarioMenu.cpp

```diff
@@ -119,15 +119,87 @@ FileMenuWidget::Prepare([[maybe_unused]] ContainerWindow &parent,
   });
 }
 
-class SystemMenuWidget final
+class ScreenRotationWidget final
   : public RowFormWidget
 {
   UI::Display &display;
   UI::EventQueue &event_queue;
 
 public:
-  SystemMenuWidget(UI::Display &_display, UI::EventQueue &_event_queue,
-                   const DialogLook &look) noexcept
+  ScreenRotationWidget(UI::Display &_display, UI::EventQueue &_event_queue,
+                 const DialogLook &look) noexcept
+    :RowFormWidget(look),
+     display(_display), event_queue(_event_queue) {}
+
+private:
+  /* virtual methods from class Widget */
+  void Prepare(ContainerWindow &parent,
+               const PixelRect &rc) noexcept override;
+};
+
+void
+ScreenRotationWidget::Prepare([[maybe_unused]] ContainerWindow &parent,
+                              [[maybe_unused]] const PixelRect &rc) noexcept
+{
+  AddButton("Lanscape", [](){
+    static constexpr const char *argv[] = {
+      "/bin/sh", "-c", 
+      "sed -ir 's/^[#]*\s*rotation=.*/rotation=0/' /boot/config.uEnv && echo The screen was rotated to landscape", 
```

Comment:
What's the point of all those `echo` calls?
If you want to display a message to the user, you should use C++ code, not shell code.

Reply:
OK, I will change it soon

Reply:
Can you give me a hint, witch function I have to use?
I think I have to use the function like:
PrettyFunctionNameHere(&dialog, "Here is your text");

Reply:
I suggest to add `echo "0" >/sys/class/graphics/fbcon/rotate_all` instead of the textual feedback. This rotates the screen immediately without the need of a reboot. And if the user sees the new orientation directly, you don't need to bother him with additional feedback (especially pop-up dialogues which are often rather annoying).

Reply:
Thanks a lot!!

Reply:
Don't write files by spawning the shell.

>  But for other echo messages I need to know which function is responsible for display messages with c++ code

The message "The screen was rotated to landscape" seems rather pointless, doesn't it? Just don't display it. The user can see the effect without you telling him.

Reply:
> The message "The screen was rotated to landscape" seems rather pointless, doesn't it? Just don't display it. The 
> user can see the effect without you telling him.

At the moment, rotating the screen only becomes active after a reboot, because only then is the config loaded. With Scumi's suggestion, the screen would be rotated directly. I'll test this with echo first, and then figure out how to do it without a shell

Reply:
> figure out how to do it without a shell

https://github.com/XCSoar/XCSoar/blob/cdd8108cd60aa1fff2f6830002fd27cc320ff0ff/src/Kobo/System.cpp#L412

Reply:
Thanks!!!

Reply:
It works! I can write strings to files. But I have one problem.

The string is always inserted in the first line. In the "/boot/config.uEnv" file, for example, there is more information, such as: 
rotation=0
LANG=de_DE.UTF-8
timeout=10

I still have to figure out how to find the right line and completely replace the old string with the new one. Can the File::WriteExisting function do this?